src/controllers/auth.controllers.js

```diff
@@ -0,0 +1,107 @@
+import { validationResult } from 'express-validator';
+import bcrypt from 'bcryptjs';
+import { generateToken } from '../utils/authentication';
+import User from '../models/user.model';
+
+export const signup = async (req, res) => {
+  const errors = validationResult(req);
+
+  if (!errors.isEmpty()) {
+    return res.status(422).json({ errors: errors.array() });
```

Comment:
there is ErrorHandler, which we use in the controllers (check how it is done in other controllers)

Reply:
Plz check the new PR